gcc/rust/ast/rust-expr.h

```diff
@@ -4729,15 +4726,47 @@ struct TupleTemplateStr
 };
 
 // Inline Assembly Node
-class InlineAsm : public ExprWithoutBlock
+class InlineAsm : private ExprWithoutBlock
 {
+private:
+  location_t locus;
+  // TODO: Not sure how outer_attrs plays with InlineAsm, I put it here in order
+  // to override, very hacky.
+  std::vector<Attribute> outer_attrs;
+
 public:
   std::vector<InlineAsmTemplatePiece> template_;
   std::vector<TupleTemplateStr> template_strs;
   std::vector<InlineAsmOperand> operands;
-  TupleClobber clobber_abi;
-  InlineAsmOptions options;
+  std::vector<TupleClobber> clobber_abi;
+  // std::set<InlineAsmOptions> options;
+  std::set<InlineAsmOptions> options;
```

Comment:
I think the `InlineAsmOptions` should be renamed to `InlineAsmOption` (no 's'), especially since we are keeping a set of them. this is very nitpicky but reading that member I thought we might be dealing with a set of sets of asm options, and I got confused

```suggestion
  std::set<InlineAsmOptions> options;
```

Reply:
resolved